### PR TITLE
Marshalize mismatch errors

### PR DESCRIPTION
**Pull request by agentdon**
Implements functionality to make MismatchErrors Marshalizeable.

### CODE REVIEW

File: lib/scientist/experiment.rb [hunk 1]
```diff
@@ -318,4 +326,12 @@ def generate_result(name)
     control = observations.detect { |o| o.name == name }
     Scientist::Result.new(self, observations, control)
   end
+
+  private
+
+  def marshalize(block)
+    def block._dump(_)
```
Comment (zerowidth) [inline lib/scientist/experiment.rb:333]: Could you add a check first before overriding? `respond_to?(:_dump) || respond_to?(:_dump_data)`

Comment (agentdon) [reply lib/scientist/experiment.rb:333]: Great idea. I'm working to add that.

Comment (agentdon) [reply lib/scientist/experiment.rb]: @zerowidth I added a check to this before overriding. Please let me know if you have any other feedback. 🙏

File: lib/scientist/experiment.rb [hunk 2]
```diff
@@ -318,4 +326,14 @@ def generate_result(name)
     control = observations.detect { |o| o.name == name }
     Scientist::Result.new(self, observations, control)
   end
+
+  private
+
+  def marshalize(block)
```
Comment (zerowidth) [inline lib/scientist/experiment.rb:332]: Can you add a comment here explaining why we want to make everything marshalable?

Comment (agentdon) [reply lib/scientist/experiment.rb:337]: Good call. Added. 👌

**Review by zerowidth — APPROVED** [review]

**Pull request merged**